ArduCopter/heli.cpp

```diff
@@ -180,4 +180,41 @@ void Copter::heli_update_rotor_speed_targets()
     rotor_runup_complete_last = motors->rotor_runup_complete();
 }
 
+#if MODE_AUTOROTATE_ENABLED == ENABLED
+// heli_update_autorotation - determines if aircraft is in autorotation and sets motors flag and switches
+// to autorotation flight mode if manual collective is not being used.
+void Copter::heli_update_autorotation()
+{
+    //Set SPDHGT_ENABLE parameter based on AROT_ENABLE parameter
+    if (arot->is_enable()  &&  !helispdhgtctrl->get_enable()) {
+        helispdhgtctrl->set_enable(1);
+    } else if (!arot->is_enable()  &&  helispdhgtctrl->get_enable()){
+        helispdhgtctrl->set_enable(0);
+    }
+
+    if (!ap.land_complete && !motors->get_interlock() && arot->is_enable()) {
+        heli_flags.in_autorotation = true;
+        // Check if an auto collective mode is being used
+        if (!flightmode->has_manual_throttle()){
+            set_mode(Mode::Number::AUTOROTATE, MODE_REASON_AUTO_AUTOROTATION);
+        }
+    } else if (ap.land_complete || (flightmode->has_manual_throttle() && motors->get_interlock())) {
+        heli_flags.in_autorotation = false;
+    }
+
+    // sets autorotation flags through out libraries
+    heli_set_autorotation(heli_flags.in_autorotation);
+    if (!ap.land_complete) {
```

Comment:
this requires a check to see if autorotation mode is enabled in order to set the bailout functionality.